<launch>
  <node pkg="px4_control"
    type="px4_control_node"
    name="px4_control_node"
    output="screen">

    <!-- Joystick Attitude Reference parameters -->
    <param name="RollMax" value="40.0"/>     <!-- degrees -->
    <param name="PitchMax" value="40.0"/>    <!-- degrees -->
    <param name="YawRateMax" value="45.0"/>  <!-- deg/sec -->
    <param name="maxThrust" value="1.0"/>    <!-- From 0 to 1 -->

    <!-- Joystick Position Reference parameters -->
    <param name="xRate" value="1.5"/>       <!-- meters/sec -->
    <param name="yRate" value="1.2"/>       <!-- meters/sec -->
    <param name="zRate" value="0.4"/>       <!-- meters/sec -->
    <param name="PosRefTimeConstant" value="0.4"/> <!-- seconds -->

    <!-- Topic with odometry information for the vehicle -->
    <param name="odomTopic" value="local_odom"/>

    <!-- Type of joystick being used. Admitted types are:
         joyXboxOne  /  joyXbox360  /  joyXbox360Wired -->
    <param name="joyDriver" value="joyXbox360Wired"/>

    <!-- Name of topic to use when handling PVA_References-->
    <param name="pvaTopic" value="px4_control/PVA_Ref_ML"/>
    
    <!-- Position Control parameters -->
    <param name="mass" value="0.7623"/>       <!-- kg -->
    <param name="gz" value="9.81"/>          <!-- m/s^2 -->
    <param name="thrustRatio" value="1.60"/>  <!-- maxThrust/Weight -->
    <param name="kpx" value="2.05"/>  <!--3.0-->
    <param name="kpy" value="1.95"/>
    <param name="kpz" value="1.75"/>  <!--1.9-->
    <param name="kvx" value="1.9"/>  <!--3.7-->
    <param name="kvy" value="1.95"/>
    <param name="kvz" value="0.95"/>  <!--1.9-->
    <param name="kix" value="0.20"/>
    <param name="kiy" value="0.20"/>
    <param name="kiz" value="0.40"/>
    <param name="maxInteg_x" value="15.0"/>
    <param name="maxInteg_y" value="15.0"/>
    <param name="maxInteg_z" value="15.0"/>
  </node>
</launch>
